MetaMask/metamask-extension
# Token allowance flow update

## Explanation
Fixes [#610](https://github.com/MetaMask/MetaMask-planning/issues/610)
### Changes introduced in this PR:

-  Updated the copy `Custom spending cap`   to `Site requested spending cap` when
-  The copies `Set a spending cap for your` and `Review the spending cap for your`are updated to `Spending cap request for your`
- The copy `Use default` is updated to  `Use site suggestion`
- The spending cap on the first page of Approve screen is pre-filled with the dapp suggested spending cap instead of having a blank input field.
-  Learn More link is added below the copy. The link is directing to https://support.metamask.io/hc/en-us/articles/6055177143579-How-to-customize-token-approvals-with-a-spending-cap.




## Screenshots/Screencaps


https://github.com/MetaMask/metamask-extension/assets/43930900/faf103e9-aff0-4178-9200-5072ffab11e6




## Manual Testing Steps
1. Create a new token from the dapp
2. Click the `Approve Token` button and make sure the token allowance page loads prefilled with the spending cap.
3. Make sure all the labels are updated based on the description.



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
